ansible/configure_galaxy.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+import argparse
+import fnmatch
+import json
+import logging
+import os
+import re
+import sys
+
+import bioblend.galaxy as bg
+
+DefaultGalaxyUrl = "http://127.0.0.1:8080"
+
+LogFormat = '%(levelname)s\t[ %(name)s ]\t%(message)s'
+Log = logging.getLogger('ConfigGalaxy')
+
+def parse_args(args):
+    parser = argparse.ArgumentParser(
+        description="Upload the workflows from a directory into the specified Galaxy server")
+
+    parser.add_argument('api_key', help="Master API key for Galaxy")
+    parser.add_argument('username', help="Username for admin user")
+    parser.add_argument('user_email', help="E-mail for admin user")
+    parser.add_argument('user_password', help="Password for admin user (at least 6 characters)")
+
+    parser.add_argument('--galaxy-url',
+                        help="Galaxy URL (default: {})".format(DefaultGalaxyUrl),
+                        default=DefaultGalaxyUrl)
+
+    parser.add_argument('--workflow-dir',
+                        help="Directory containing workflow files to be uploaded (must have .ga extension; default: '.')",
+                        default='.')
+
+    options = parser.parse_args(args)
+
+    Log.debug("Galaxy config running from %s", os.path.abspath('.'))
+    if len(options.user_password) < 6:
+        parser.error("Password needs to be at least 6 characters long.  User not added!")
+    if not re.match(r'[^@]+@[^@]+\.[^@]+', options.user_email):
+        parser.error("Email address has an invalid syntax.  User not added!")
+    if not os.path.isdir(options.workflow_dir) or not os.access(options.workflow_dir, os.R_OK | os.X_OK):
```

Comment:
I can change it to set a user and no workflows, but not the other way around I'm afraid.

There it no real user account associated with the "master" api key so some operations can't be performed.  Among these is workflow installation.  This means that  we need to set up the specified user and create an API key for the account to install the workflows.

This same issue makes it a little messy to separate the steps into multiple scripts.  I'd have to output the (potentially new) admin user's API key from the first script (I suppose by writing it to stdout), parse that output and pass it into the second script.  I think it's simpler and more robust to keep all galaxy configuration steps in the same Python script, where it's easier to control how we pass things around.

In conclusion, I'm thinking of resolving this by adding a `--skip-steps=...` option that can accept a list of arguments.  For now the only possible step to skip is workflow installation; in the future more may be possible.  In all cases, user set-up is mandatory.  What do you think?